src/main/java/net/openhft/chronicle/wire/GenerateMethodReader.java

```diff
@@ -260,20 +263,14 @@ private void generateSourceCode() {
         sourceCode.append(eventNameSwitchBlock);
 
         sourceCode.append("default:\n" +
-                "valueIn.skipValue();\n" +
```

Comment:
I suspect the code above is incorrect now
```
        sourceCode.append("default:\n" +
                "valueIn.skipValue();\n" +
                "return false;\n" +
                "}\n" +
                "}\n" +
                "else {\n" +
                "lastEventName = wireIn.readEvent(String.class);\n" +
                "}\n" +
                "try {\n" +
                "if (Jvm.isDebug())\n" +
                "debugLoggingParselet.accept(lastEventName, valueIn);\n" +
                "if (lastEventName == null)\n" +
                "throw new IllegalStateException(\"Failed to read method name or ID\");\n" +
                "switch (lastEventName) {\n" +
                "case MethodReader.HISTORY:\n" +
                "valueIn.marshallable(messageHistory);\n" +
                "break;\n\n");
```
If we detect `BinaryWireCode.FIELD_NUMBER` and we don't find an appropriate method ID in the switch block, we still return `false` hoping that a delegate will handle this case for us.
Looking at `VanillaMethodReader`, I think we should consume the unexpected event with `WireParser#SKIP_READABLE_BYTES` and return true.

Reply:
Thanks - let me review again (and add some tests)

Reply:
@glukos please re-review!